docs/flow/modin/experimental/core/execution/native/implementations/omnisci_on_native/index.rst

```diff
@@ -1,7 +1,237 @@
 :orphan:
 
+OmniSciOnNative execution
+=========================
+
+OmniSciDB is an open-source SQL-based relational database designed for the
+massive parallelism of modern CPU and GPU hardware. Its execution engine
+is built on LLVM JIT compiler.
+
+OmniSciDB can be embedded into an application as a dynamic library that
+provides both C++ and Python APIs. A specialized in-memory storage layer
+provides an efficient way to import data in Arrow table format.
+
+`OmniSciOnNative` execution uses OmniSciDB for both as a storage format and for
+actual data transformation.
+
+Relational engine limitations
+-----------------------------
+
+Using a relational database engine implies a set of restrictions on
+operations we can execute on a dataframe.
+
+1. We cannot handle frames that use data types not supported by OmniSciDB.
+Currently, we allow only integer, float, string, and categorical data types.
+
+2. Column data should be homogeneous.
+
+3. Can only support operations that map to relational algebra. This means
+most operations are supported over a single axis (axis=0) only. Non-relational
+operations like transposition and pivot are not supported.
+
+When the unsupported data type is detected or unsupported operations is requested
+we use the original pandas framework.
+
+Partitions
+----------
+
+Partitioning is used to achieve high parallelism. In the case of OmniSciDB
+based execution parallelism is provided by OmniSciDB execution engine
```

Comment:
```suggestion
In Modin, partitioning is used to achieve high parallelism. In case of OmniSciDB-based
execution parallelism is provided by OmniSciDB execution engine itself
```